Compression test don't compile without zstd

The compression test should imo either use the non optional lzma or just not be run at all if zstd is not enabled (or just run for all enabled compression algos)
Affected Version: 6.1.6
```
FAILED: test/9f86d08@@compression@exe/compression.cpp.o 
g++ -Itest/9f86d08@@compression@exe -Iinclude -I../include -Isrc -I../src -flto -fdiagnostics-color=always -DNDEBUG -pipe -std=c++11 -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pthread -DGTEST_HAS_PTHREAD=1 -MD -MQ 'test/9f86d08@@compression@exe/compression.cpp.o' -MF 'test/9f86d08@@compression@exe/compression.cpp.o.d' -o 'test/9f86d08@@compression@exe/compression.cpp.o' -c ../test/compression.cpp
../test/compression.cpp:62:53: error: 'ZSTD_INFO' was not declared in this scope; did you mean 'ZIP_INFO'?
   62 | using CompressionAlgo = ::testing::Types<LZMA_INFO, ZSTD_INFO>;
      |                                                     ^~~~~~~~~
      |                                                     ZIP_INFO
../test/compression.cpp:62:62: error: template argument 2 is invalid
   62 | using CompressionAlgo = ::testing::Types<LZMA_INFO, ZSTD_INFO>;
      |                                                              ^
In file included from /usr/include/gtest/gtest.h:67,
                 from ../test/compression.cpp:46:
../test/compression.cpp:64:34: error: 'CompressionAlgo' was not declared in this scope; did you mean 'CompressionTest'?
   64 | TYPED_TEST_CASE(CompressionTest, CompressionAlgo);
      |                                  ^~~~~~~~~~~~~~~
../test/compression.cpp:64:1: error: template argument 1 is invalid
   64 | TYPED_TEST_CASE(CompressionTest, CompressionAlgo);
      | ^~~~~~~~~~~~~~~
../test/compression.cpp:64:1: error: expected initializer before 'gtest_type_params_CompressionTest_'
   64 | TYPED_TEST_CASE(CompressionTest, CompressionAlgo);
      | ^~~~~~~~~~~~~~~
../test/compression.cpp:66:1: error: 'gtest_type_params_CompressionTest_' was not declared in this scope; did you mean 'gtest_type_params_CompressionTest_NameGenerator'?
   66 | TYPED_TEST(CompressionTest, compress) {
      | ^~~~~~~~~~
../test/compression.cpp:66:1: error: template argument 3 is invalid
   66 | TYPED_TEST(CompressionTest, compress) {
      | ^~~~~~~~~~
../test/compression.cpp:66:1: error: 'gtest_type_params_CompressionTest_' was not declared in this scope; did you mean 'gtest_type_params_CompressionTest_NameGenerator'?
   66 | TYPED_TEST(CompressionTest, compress) {
      | ^~~~~~~~~~
../test/compression.cpp:66:1: error: no matching function for call to 'GenerateNames<{anonymous}::gtest_type_params_CompressionTest_NameGenerator, <expression error> >()'
   66 | TYPED_TEST(CompressionTest, compress) {
      | ^~~~~~~~~~
In file included from /usr/include/gtest/gtest.h:59,
                 from ../test/compression.cpp:46:
/usr/include/gtest/internal/gtest-internal.h:634:26: note: candidate: 'template<class NameGenerator, class Types> std::vector<std::__cxx11::basic_string<char> > testing::internal::GenerateNames()'
  634 | std::vector<std::string> GenerateNames() {
      |                          ^~~~~~~~~~~~~
/usr/include/gtest/internal/gtest-internal.h:634:26: note:   template argument deduction/substitution failed:
In file included from /usr/include/gtest/gtest.h:67,
                 from ../test/compression.cpp:46:
../test/compression.cpp:66:1: error: template argument 2 is invalid
   66 | TYPED_TEST(CompressionTest, compress) {
      | ^~~~~~~~~~
```

--- a/test/compression.cpp
+++ b/test/compression.cpp
@@ -59,7 +59,15 @@ class CompressionTest : public testing::Test {
     typedef zim::Uncompressor<T> DecompressorT;
 };
 
-using CompressionAlgo = ::testing::Types<LZMA_INFO, ZSTD_INFO>;
+using CompressionAlgo = ::testing::Types<
+  LZMA_INFO
+#if defined(ENABLE_ZSTD)
+  ,ZSTD_INFO
+#endif
+#if defined(ENABLE_ZLIB)
+  ,ZIP_INFO
+#endif
+>;
 
 TYPED_TEST_CASE(CompressionTest, CompressionAlgo);
 

